src/linear_algebra/basic.lean

```diff
@@ -942,6 +942,9 @@ lemma span_attach_bUnion [decidable_eq M] {α : Type*} (s : finset α) (f : s 
   span R (s.attach.bUnion f : set M) = ⨆ x, span R (f x) :=
 by simpa [span_Union]
 
+lemma sup_span : p ⊔ span R s = span R (p ∪ s) :=
+by rw [submodule.span_union, p.span_eq]
```

Comment:
If we're going to have this lemma, we should probably have `span_sup` too

Reply:
I agree we should have both or neither, so I've added `span_sup` also for now.